src/data_gradients/feature_extractors/common/image_duplicates.py

```diff
@@ -0,0 +1,264 @@
+from collections import Iterable
+from typing import List
+from data_gradients.common.registry.registry import register_feature_extractor
+from data_gradients.feature_extractors.abstract_feature_extractor import AbstractFeatureExtractor
+from data_gradients.feature_extractors.abstract_feature_extractor import Feature
+from imagededup.methods import DHash
+
+import os
+
+from data_gradients.utils.data_classes import ImageSample
+
+
+@register_feature_extractor()
+class ImageDuplicates(AbstractFeatureExtractor):
+    """
+    Extracts image duplicates, in the directories train_image_dir, valid_image_dir (when present) and their intersection.
+
+    Under the hood, uses Difference Hashing (http://www.hackerfactor.com/blog/index.php?/archives/529-Kind-of-Like-That.html)
+     and considers duplicates if and only if they have the exact same hash code. This means that regardless of color format
+      (i.e BGR, RGB grreyscale) duplicates will be found, but might result (rarely) in false positives.
```

Comment:
`grreyscale` -> `grayscale` (or `greyscale` if you prefer)

Reply:
fixed in latest commit.